index.js

```diff
@@ -12,25 +13,37 @@ var isPromise = require('is-promise');
  */
 
 module.exports = function (func, cb) {
-  var async = false;
-  cb = once(cb);
+  return function () {
+    var async = false;
+    var promise = null;
+    cb = cb && once(cb);
+    if (typeof Promise !== 'undefined') {
+      promise = new Promise(function (resolve, reject) {
+        cb = promiseResolver(resolve, reject, cb);
+      });
+    } else if (!cb) {
+      throw new Error('no callback or global Promise implementation');
```

Comment:
12.7 also works (not sure on 12.0-12.6).

You could also make it work on 0.10 by setting global.Promise to bluebird, etc.

How about:
"You need to provide a callback or upgrade to Node 0.12.?? if you intend to use Promises" 

We could discuss hacking 0.10 to work in the documentation